include/miral/miral/display_configuration.h

```diff
@@ -41,6 +45,25 @@ class ConfigurationOption;
 class DisplayConfiguration
 {
 public:
+    /// A class providing access to an arbitrary piece of data from the [DisplayConfiguration].
+    /// This is specifically useful when a user wants to extend the based display configuration
+    /// with some sort of custom payload (e.g. a user may want to extend the layout configuration
+    /// with information describing the position and size of specific applications).
+    /// \remark Since MirAL 5.3
+    class Node
+    {
+    public:
+        explicit Node(YAML::Node const& node);
+        auto as_string() const -> std::optional<std::string>;
+        auto as_int() const -> std::optional<int>;
+        void for_each(std::function<void(Node const&)> const& f) const;
+        auto at(std::string const& key) const -> std::optional<Node>;
```

Comment:
This still looks harder to use than having an explicit type

```suggestion
        auto type() const -> Type;
        auto as_string() const -> std::string  pre (type() == string);
        auto as_int() const -> int pre (type() == integer);
        void for_each(std::function<void(Node const&)> const& f) const pre (type() == sequence);
        auto at(std::string const& key) const -> Node pre (type() == dictionary);
```